cirq-google/cirq_google/devices/google_noise_properties_test.py

```diff
@@ -0,0 +1,258 @@
+    q0, q1 = cirq.LineQubit.range(2)
+    props = sample_noise_properties([q0, q1], [(q0, q1), (q1, q0)])
+    model = NoiseModelFromGoogleNoiseProperties(props)
+    circuit = cirq.Circuit(op)
+    noisy_circuit = circuit.with_noise(model)
+    assert len(noisy_circuit.moments) == 4
+    assert len(noisy_circuit.moments[0].operations) == 1
+    assert noisy_circuit.moments[0].operations[0] == op.with_tags(PHYSICAL_GATE_TAG)
+
+    # Depolarizing noise
+    assert len(noisy_circuit.moments[1].operations) == 1
+    depol_op = noisy_circuit.moments[1].operations[0]
+    assert isinstance(depol_op.gate, cirq.DepolarizingChannel)
+    assert np.isclose(depol_op.gate.p, 0.00719705)
+
+    # FSim angle corrections
+    assert len(noisy_circuit.moments[2].operations) == 1
+    fsim_op = noisy_circuit.moments[2].operations[0]
+    assert isinstance(fsim_op.gate, cirq.PhasedFSimGate)
+    assert fsim_op == PhasedFSimGate(theta=0.01, zeta=0.03, chi=0.04, gamma=0.05, phi=0.02).on(
+        q0, q1
+    )
+
+    # Thermal noise
+    assert len(noisy_circuit.moments[3].operations) == 2
+    thermal_op_0 = noisy_circuit.moments[3].operation_at(q0)
+    thermal_op_1 = noisy_circuit.moments[3].operation_at(q1)
+    assert isinstance(thermal_op_0.gate, cirq.KrausChannel)
+    assert isinstance(thermal_op_1.gate, cirq.KrausChannel)
+    thermal_choi_0 = cirq.kraus_to_choi(cirq.kraus(thermal_op_0))
+    thermal_choi_1 = cirq.kraus_to_choi(cirq.kraus(thermal_op_1))
+    expected_thermal_choi = np.array(
+        [
+            [1, 0, 0, 9.99680051e-01],
+            [0, 3.19948805e-04, 0, 0],
+            [0, 0, 0, 0],
+            [9.99680051e-01, 0, 0, 9.99680051e-01],
+        ]
+    )
+    assert np.allclose(thermal_choi_0, expected_thermal_choi)
+    assert np.allclose(thermal_choi_1, expected_thermal_choi)
+
+
+def test_supertype_match():
+    # Verifies that ops in gate_pauli_errors which only appear as their
+    # supertypes in fsim_errors are properly accounted for.
+    q0, q1 = cirq.LineQubit.range(2)
+    op_id = OpIdentifier(cirq_google.SycamoreGate, q0, q1)
+    test_props = sample_noise_properties([q0, q1], [(q0, q1), (q1, q0)])
+    expected_err = test_props._depolarizing_error[op_id]
+
+    props = sample_noise_properties([q0, q1], [(q0, q1), (q1, q0)])
+    props.fsim_errors = {
+        k: cirq.PhasedFSimGate(0.5, 0.4, 0.3, 0.2, 0.1)
+        for k in [OpIdentifier(cirq.FSimGate, q0, q1), OpIdentifier(cirq.FSimGate, q1, q0)]
+    }
+    assert props._depolarizing_error[op_id] != expected_err
+
+
+def test_measure_gates():
+    q00, q01, q10, q11 = cirq.GridQubit.rect(2, 2)
+    qubits = [q00, q01, q10, q11]
+    props = sample_noise_properties(
+        qubits,
+        [
+            (q00, q01),
+            (q01, q00),
+            (q10, q11),
+            (q11, q10),
+            (q00, q10),
+            (q10, q00),
+            (q01, q11),
+            (q11, q01),
+        ],
+    )
+    model = NoiseModelFromGoogleNoiseProperties(props)
```

Comment:
please write a short comment explaining what the noisy circuit should look like.

Reply:
Done - added before the circuit-length check.